src/libai/nn/NeuralNetwork.java

```diff
@@ -67,21 +80,30 @@ public void setProgressBar(ProgressDisplay pb) {
 	 * @param length	How many patterns will be used.
 	 * @param minerror	The minimal error expected.
 	 */
-	public abstract void train(Matrix[] patterns, Matrix[] answers, double alpha, int epochs, int offset, int length, double minerror);
+	public void train(Matrix[] patterns, Matrix[] answers, double alpha, int epochs, int offset, int length, double minerror) {
```

Comment:
I could actually write a separate `protected void NeuralNetwork#validate(...)` keeping `train(...)` abstract. 
About the minerror effect it's a matter of calling `super.validate(..., -minerror,...)`.

(I'm currently writing from my phone, but in a couple of days I'll be back at home and submit the commits)